standalone-metastore/metastore-client/src/main/java/org/apache/hadoop/hive/metastore/client/BaseMetaStoreClient.java

```diff
@@ -557,6 +561,32 @@ public final List<Partition> dropPartitions(String catName, String dbName, Strin
     return dropPartitions(catName, dbName, tblName, partExprs, options, null);
   }
 
+  @Override
+  public List<Partition> dropPartitions(String catName, String dbName, String tblName,
+      List<Pair<Integer, byte[]>> partExprs, PartitionDropOptions options, EnvironmentContext context)
+      throws NoSuchObjectException, MetaException, TException {
+    if (context == null) {
+      context = new EnvironmentContext();
+    }
+
+    if (context.getProperties() != null &&
+        Boolean.parseBoolean(context.getProperties().get(SKIP_DROP_PARTITION))) {
```

Comment:
Should we check the `SKIP_DROP_PARTITION` in all the clients new signature, not only `ThriftHiveMetastoreClient`?

Reply:
Currently all clients will use the `ThriftHiveMetastoreClient` as the delegate, so it's OK to just check `SKIP_DROP_PARTITION` in `ThriftHiveMetastoreClient`.

Reply:
in that case should we drop this check from the BaseMetastoreClient?

Reply:
Addressed.